components/Modals/PaymentOptionModal.tsx

```diff
@@ -0,0 +1,152 @@
+        disabled && "bg-pine-100 hover:border-pine-100"
+      )}
+      onClick={!disabled ? onClick : undefined}
+    >
+      <div className="flex py-2">
+        <Image alt={name} src={icon} height={25} width={25} />
+
+        <h3 className="px-3">{name}</h3>
+      </div>
+      <div className="flex">
+        <Double className="min-w-[15%]" label="Fees" value={`${fees}%`} />
+        <Double className="min-w-[40%]" label="Works for" value={worksFor} />
+        <Double className="min-w-[40%]" label="Delay" value={delay} />
+      </div>
+    </div>
+  );
+
+  return (
+    <div className="flex flex-col max-w-[343px] text-pine-900 p-2">
+      <div className="flex flex-row justify-between p-3">
+        <div></div>
+        <div className="border-2 rounded-full border-cyan-400 px-3">
+          🔗 {sliceAddress(address!)}
+        </div>
+        <button onClick={() => closeModal()}>
+          <Image alt="x" src="/x.svg" height={16} width={16} />
+        </button>
+      </div>
+      <h2 className="text-center">Choose a payment option to buy Glo Dollar</h2>
+      {isConnected && address && (
+        <>
+          <BuyBox
+            name="Uniswap"
+            icon="/uniswap.png"
+            fees=".01"
+            worksFor="🔐 Crypto"
+            delay="⚡ Instant"
+            onClick={() => buyWithUniswap(1000)}
+          />
+          <BuyBox
+            name="Ratio"
+            icon="/ratio.png"
+            fees="0"
+            worksFor="🇺🇸 US citizens"
+            delay="Up to 3 days"
+            disabled={loading}
+            onClick={() => {
+              const parent = document.getElementById("ratio-button-parent");
+              const button = parent?.firstChild as HTMLButtonElement;
+              if (button) {
+                setLoading(true);
+                button.click();
+
+                // The only workaround to handle Ratio modal position
+                // Close our modal after Ratio modal is detected
+                const tryClosingModal = () => {
+                  const el = document.getElementById("radix-:r3:");
+                  if (el) {
+                    closeModal();
+                  } else {
+                    setTimeout(() => {
+                      tryClosingModal();
+                    }, 1000);
+                  }
+                };
+
+                tryClosingModal();
+              }
+            }}
+          />
+          <BuyBox
+            name="Transak"
+            icon="/transak.png"
+            fees="1-5"
+            worksFor="🌍 world"
+            delay="⚡ Instant"
+            onClick={() => buyWithTransak(1000, address)}
+          />
+          <BuyBox
+            name="Coinbase"
+            icon="/coinbase.png"
+            fees="0-3"
+            worksFor="🌍 world"
+            delay="⚡ Instant"
+            onClick={() => buyWithCoinbase()}
+          />
+        </>
+      )}
+      <button className="bg-pine-300 h-[52px] py-3.5 mx-6">
```

Comment:
should we comment out if the screens are TBD for EthCC?

Reply:
@gglucass ?